docs/decisions/0013-custom-groups-for-enterprise-learners.rst

```diff
@@ -0,0 +1,155 @@
+------
+- uuid, name, created, modified, history (boilerplate)
+- enterprise_customer_uuid (NOT NULL, FK to EnterpriseCustomer, related_name=”groups”)
+- policy_uuid (UUIDField)
+
+**CRUD**
+------
+**api/v1/enterprise-group/<group_uuid>**
+
+Outputs
+==========
+The root URL for getting the basic information about the group
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'group_uuid': 'group_uuid',
+               'name': 'group_name',
+               'enterprise_customer_uuid': 'enterprise_customer_uuid',
+               'policy_uuid': 'policy_uuid'
+           }
+       ]
+   }
+
+
+**api/v1/enterprise-group/?learner_uuid=[enterprise_customer_user_id]&enterprise_uuid=[enterprise_uuid]**
+
+Accepted Query Params
+==========
+- ``learner_uuid`` (optional): Get all the groups that the learner is associated with 
+- ``enterprise_uuid`` (optional): Get all the groups under the enterprise
+
+Outputs
+==========
+Returns a paginated list of groups filtered by the query params
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'group_uuid': 'group_uuid',
+               'name': 'group_name',
+               'enterprise_customer_uuid': 'enterprise_customer_uuid',
+               'policy_uuid': 'policy_uuid'
+           }
+       ]
+   }
+
+
+**GET (list) /learners**
+------
+**api/v1/enterprise-group/<group_uuid>/learners/**
+
+Outputs
+==========
+Returns a paginated list of learners that are associated with the enterprise group uuid 
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'learner_uuid': 'enterprise_customer_user_id',
+               'enterprise_group_membership_uuid': 'enterprise_group_membership_uuid',
+           }
+       ]
+   }
+
+
+**POST /assign_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/assign_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to assign to the group
+
+Outputs
+==========
+Returns a list of the EnterpriseGroupMembership objects that were created 
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'learner_uuid': 'enterprise_customer_user_id',
+               'enterprise_group_membership_uuid': 'enterprise_group_membership_uuid',
+           }
+       ]
+   }
+
+
+**POST /remind_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/remind_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to remind of their group membership
+
+
+**POST /remove_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/remove_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to assign to the group
```

Comment:
[Question] do we want to return the membership uuids of the learners removed?

Reply:
I don't know I don't really see the point of it? What use could we have of them?

Reply:
it gives consumers more information on the exact records that were modified as a result of their request but I don't think it's fully necessary for the operations we're looking to perform for MVP